src/librustc/mir/interpret/error.rs

```diff
@@ -321,9 +321,11 @@ pub enum UndefinedBehaviorInfo {
     RemainderByZero,
     /// Overflowing inbounds pointer arithmetic.
     PointerArithOverflow,
+    /// Invalid metadata in a wide pointer (using `str` to avoid allocations).
+    InvalidMeta(&'tcx str),
```

Comment:
```suggestion
    InvalidMeta(&'static str),
```

Reply:
I see no good reason to require the static lifetime here, and we are already using `&'tcx str` in another error type.

Reply:
Is there any way to make an `&'tcx str` that isn't an `&'static str`?

Reply:
I'm not sure. I also don't think it matters. What is the advantage of using the more restrictive `static` lifetime?

Reply:
It means one less lifetime to add everywhere.

Reply:
"everywhere" is like 2 lines here. `InterpError` needs that lifetime anyway.

Reply:
I do prefer `'static` over `'tcx` to communicate that this is not an interned string but either a leaked one or a true literal

Reply:
All right, I'll change the lifetime.